### PR TITLE
V1 test cased imported to V2 and Search E2E

### PR DESCRIPTION
**Pull request by tareq89**
## Description

Issue link: (https://github.com/opencrvs/opencrvs-core/issues/9435)

core pr: https://github.com/opencrvs/opencrvs-core/pull/9586
Country config PR: https://github.com/opencrvs/opencrvs-countryconfig/pull/740

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

**Comment by adskyiproger** [comment]
Your environment is deployed to https://ocrvs-9435.opencrvs.dev.

File: e2e/testcases/v2-advanced-search/1-search-birth-event-declaration-registration-details.spec.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    const record: Awaited<ReturnType<typeof createDeclaration>> =
+      await createDeclaration(token, {
+        'child.dob': faker.date
+          // Randomly chosen DOB between 2010-01-01 and 2020-12-31
+          // Ensures the created record appears on the first page of search results
+          .between({ from: '2010-01-01', to: '2020-12-31' })
+          .toISOString()
+          .split('T')[0],
+        //@ts-ignore
```
Comment (cibelius) [inline e2e/testcases/v2-advanced-search/1-search-birth-event-declaration-registration-details.spec.ts]: We should be able to improve the typing instead of using `// @ts-ignore`

File: e2e/testcases/v2-advanced-search/1-search-birth-event-declaration-registration-details.spec.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+/**
+ * Converts a numeric month value (1–12) to its corresponding short English month name (e.g., "Jan", "Feb").
+ *
+ * @param {number} month - The month number (1 for January, 12 for December).
+ * @returns {string} The short name of the month in English.
+ *
+ * @example
+ * getMonthShortName(1); // returns "Jan"
+ * getMonthShortName(12); // returns "Dec"
+ */
```
Comment (cibelius) [inline e2e/testcases/v2-advanced-search/1-search-birth-event-declaration-registration-details.spec.ts]: This comment doesnt seem right

File: e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+          // Randomly chosen DOB between 2010-01-01 and 2020-12-31
+          // Ensures the created record appears on the first page of search results
+          .between({ from: '2010-01-01', to: '2020-12-31' })
+          .toISOString()
+          .split('T')[0],
+        //@ts-ignore
+        'child.reason': 'Other', // needed for late dob value
+        'child.gender': 'female'
+      })
+    ;[yyyy, mm, dd] = record.declaration['child.dob'].split('-')
```
Comment (cibelius) [inline e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts:47]: weird line, should be on separate line?

File: e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    fullNameOfChild =
```
Comment (cibelius) [inline e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts]: you can use template literal, e.g. `${record.declaration['child.firstname]} ${record.declaration['child.surname']}` 

there also exists a `formatName()` helper which probably could be used here

**Review by cibelius — COMMENTED** [review]
do any of these tests test for the actual search results, or is that not implemented yet?

File: e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts (hunk 3 above)
Comment (tareq89) [reply e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts:47]: This is called defensive semicolon, we have defined yyyy,mm,dd in outer scope so that we can access it in other sibling scopes. This syntax lets us to use previously defined variable assignments in a destructuring manner.

**Comment by tareq89** [comment]
> do any of these tests test for the actual search results, or is that not implemented yet?

Actual search results. For example, it test for if childFullName is available in search result page, search result count etc.

**Review by cibelius — APPROVED** [review]

File: e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts (hunk 3 above)
Comment (cibelius) [reply e2e/testcases/v2-advanced-search/2-search-birth-event-declaration-child-details.spec.ts:47]: gothca 👍 im not a big fan since it doesnt feel that intuitive/readable to me, but thats just my opinion :)
